lib/druid/data_source.rb

```diff
@@ -14,12 +15,21 @@ def initialize(name, uri)
       end
     end
 
-    def metadata
-      @metadata ||= metadata!
+    def metadata(opts = {})
+      @metadata ||= metadata!(opts)
```

Comment:
Good point, @hollow I think since this will likely be an edge case, only having it on `metadata!` should be fine.